File: examples/php/resolve.php
<?php
require __DIR__ . '/vendor/autoload.php';

use Net_DNS2\Resolver;

$domain = '8.8.8.8.geons';
$resolver = new Resolver([
    'nameservers' => ['127.0.0.1'],
    'port' => 5300,
]);

try {
    $response = $resolver->query($domain, 'TXT');
    foreach ($response->answer as $record) {
        if ($record->type === 'TXT') {
            echo implode('', $record->text) . PHP_EOL;
        }
    }
} catch (Exception $e) {
    fwrite(STDERR, 'DNS lookup error: ' . $e->getMessage() . "\n");
    exit(1);
}

File: examples/python/resolve.py
#!/usr/bin/env python3

import dns.resolver

resolver = dns.resolver.Resolver(configure=False)
resolver.nameservers = ["127.0.0.1"]
resolver.port = 5300

answers = resolver.resolve("8.8.8.8.geons", "TXT")
for answer in answers:
    print(answer.to_text())
